### PR TITLE
Feature/issue 213

### PR DESCRIPTION
**Pull request by crhowell**
Updates `_extract_captions_json` method to do additional html splitting to attempt to extract out the `playabilityStatus` JSON object so we may match and raise a more specific error against the "status" field.

Updates `CouldNotRetrieveTranscript` class to accept `playabilityStatus` JSON object so we can generate more informative errors with cause, reason, subreason attached. 

Adds two playability error helper methods: `get_playability_subreason`, `get_playability_error`

Adds tests to cover the 2 new error classes: `VideoUnplayable` and `LoginRequired`

### CODE REVIEW

File: youtube_transcript_api/_errors.py [hunk 1]
```diff
@@ -118,3 +134,46 @@ def cause(self):
...
+        return ""
+    
+    subreason = renderer.get("subreason", dict()).get("runs", list())
+    
+    if not subreason:
+        return ""
+    
+    if len(subreason) > 0:
+        return subreason[0]['text']
+    return ""
```
Comment (ilystsov) [inline youtube_transcript_api/_errors.py:179]:
```suggestion
    error_screen = playability_json.get("errorScreen", {})
    renderer = error_screen.get("playerErrorMessageRenderer", {})
    subreason = renderer.get("subreason", {}).get("runs", [])

    if subreason:
        return subreason[0].get('text', '')

    return ""
```

**Review by ilystsov — CHANGES_REQUESTED** [review]
Remove redundant checks, simplify the code

File: youtube_transcript_api/_errors.py [hunk 2]
```diff
@@ -118,3 +134,38 @@ def cause(self):
...
+def get_playability_error(playability_json):
+    """
+    Using the json extracted from playabilityStatus,
+    returns a custom error based on the value of the "status" key.
+    
+    Anything that is not {"status": "OK"} is likely an error.
+    """
+    reason = playability_json.get("status")
+    if reason == 'LOGIN_REQUIRED':
+        # error for age related playability
```
Comment (AndreCAndersen) [inline youtube_transcript_api/_errors.py:148]: `LOGIN_REQUIRED` also happens when you need to "Sign in to confirm you’re not a bot". This simpler pull request addresses this: https://github.com/jdepoix/youtube-transcript-api/pull/337

**Comment by jdepoix** [comment]
Hi @crhowell !

I am so sorry for taking so long to make any progress on this, but I didn't want to merge this without putting more time into testing it out and finding different edge cases! 🙈 

I have finally gotten around to putting some time into it and have now refactored how the errors are parsed, which has allowed me to add a few new exception types, that should make quite a few error cases a lot more explicit! I have released this as part of [v1.0.0](https://github.com/jdepoix/youtube-transcript-api/releases/tag/v1.0.0). Since I have refactored some interfaces as part of this release I unfortunately I wasn't able to directly merge your code. But although I will have close this PR without merging it, I thank you very much for your contribution, as it was a very helpful resource while refactoring the error parsing! 😊🙏

**Comment by crhowell** [comment]
> Hi @crhowell !
> 
> I am so sorry for taking so long to make any progress on this, but I didn't want to merge this without putting more time into testing it out and finding different edge cases! 🙈
> 
> I have finally gotten around to putting some time into it and have now refactored how the errors are parsed, which has allowed me to add a few new exception types, that should make quite a few error cases a lot more explicit! I have released this as part of [v1.0.0](https://github.com/jdepoix/youtube-transcript-api/releases/tag/v1.0.0). Since I have refactored some interfaces as part of this release I unfortunately I wasn't able to directly merge your code. But although I will have close this PR without merging it, I thank you very much for your contribution, as it was a very helpful resource while refactoring the error parsing! 😊🙏

@jdepoix  No worries at all, glad you were able to find time to work in a refactor. My work has kept me fairly busy and working quite a lot in Golang the past few years. I do miss Python from time to time but Golang is great too, love the tooling.
